src/shared/find-esp.c

```diff
@@ -367,39 +368,34 @@ static int verify_esp(
          *  -EACESS        → if 'unprivileged_mode' is set, and we have trouble accessing the thing
          */
 
-        relax_checks =
-                getenv_bool("SYSTEMD_RELAX_ESP_CHECKS") > 0 ||
-                FLAGS_SET(flags, VERIFY_ESP_RELAX_CHECKS);
+        /* Non-root user can only check the status, so if an error occurred in the following, it does not
+         * cause any issues. Let's also, silence the error messages. */
 
-        /* Non-root user can only check the status, so if an error occurred in the following, it does not cause any
-         * issues. Let's also, silence the error messages. */
-
-        if (!relax_checks) {
-                struct statfs sfs;
-
-                if (statfs(p, &sfs) < 0)
-                        /* If we are searching for the mount point, don't generate a log message if we can't find the path */
-                        return log_full_errno((searching && errno == ENOENT) ||
-                                              (unprivileged_mode && errno == EACCES) ? LOG_DEBUG : LOG_ERR, errno,
-                                              "Failed to check file system type of \"%s\": %m", p);
+        if (getenv_bool("SYSTEMD_RELAX_ESP_CHECKS") > 0 || FLAGS_SET(flags, VERIFY_ESP_RELAX_CHECKS))
+                /* If relaxed checks are requested, don't require the esp directory to be rhe root of the
+                 * filesystem so that image builders can install the bootloader to a regular directory before
+                 * packing that directory up into its own partition. */
+                goto finish;
 
-                if (!F_TYPE_EQUAL(sfs.f_type, MSDOS_SUPER_MAGIC))
-                        return log_full_errno(searching ? LOG_DEBUG : LOG_ERR,
-                                              SYNTHETIC_ERRNO(searching ? EADDRNOTAVAIL : ENODEV),
-                                              "File system \"%s\" is not a FAT EFI System Partition (ESP) file system.", p);
-        }
+        if (statfs(p, &sfs) < 0)
+                /* If we are searching for the mount point, don't generate a log message if we can't
+                        * find the path */
+                return log_full_errno((searching && errno == ENOENT) ||
+                                        (unprivileged_mode && errno == EACCES) ? LOG_DEBUG : LOG_ERR, errno,
+                                        "Failed to check file system type of \"%s\": %m", p);
```

Comment:
Indentation. Also the line break in the middle of the ternary expression seems strange. Why not that on one line, and then `errno,` on the next?